dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/WSRdbManagedConnectionImpl.java

```diff
@@ -485,8 +491,33 @@ else if (!DataSource.class.equals(mcf.type) && !Driver.class.equals(mcf.type) &&
             cri = WSConnectionRequestInfoImpl.createChangableCRIFromNon(cri);
         cri.setDefaultValues(defaultCatalog, defaultHoldability, defaultReadOnly, defaultTypeMap, defaultSchema, defaultNetworkTimeout);
 
-
-        //  - synchronize the connectionproperties
+        // Check to make sure a data source configured with 'NONE (0)' is compatible with the driver being used. 
+        if(config.isolationLevel == Connection.TRANSACTION_NONE) {
+            try {
+                if(conn.getTransactionIsolation() == Connection.TRANSACTION_NONE) {
+                    if(isTraceOn && tc.isDebugEnabled()) {
+                        Tr.debug(this, tc, "DataSource configured with an isolation level of 'NONE (0)' and the driver's isolation level is already 'NONE (0)'."); 
+                    }
+                    useDriverIsoLvl = true; //Expected behavior
+                } else {
+                    //attempt to set isolation level of the driver to 'NONE (0)'
```

Comment:
java.sql.Connection.setTransactionIsolationLevel clearly states that TRANSACTION_NONE cannot be supplied as a parameter.  Why are we going out of our way to attempt what the JDBC API itself says cannot be done?  Is there a particular JDBC driver we are trying to accommodate here?  If not, we should remove this path that deviates from the JDBC spec unless/until there is shown to be a pressing need for it.